### ISSUE TITLE
CheckDataRegression() - continuous outcome transformation leads to rank regression

### ISSUE THREAD

**Issue by ruweng** [barbarabodinier/sharp#6]
In CheckDataRegression(), all one-column ydata (even continuous ydata) is currently transformed into factors and later changed back to numeric (subtracting 1 for the reference category).

```
   # Turning matrix with one column into vector
    if (is.matrix(ydata)) {
      if (ncol(ydata) == 1) {
        ydata <- as.vector(ydata)
      }
    }
    # Turning vector into factor
    if (is.vector(ydata)) {
      ydata <- as.factor(ydata)
    }

    # Defining reference category and final data type
    if (is.factor(ydata)) {

      [...]

      ydata <- as.numeric(ydata) - 1
      ydata <- matrix(ydata, ncol = 1)
    }
  }
```

This effectively changes a continuous outcome to the rank of the continuous outcome (minus 1). (see pictures) 

<img width="693" alt="Screenshot 2023-10-27 at 16 02 56" src="https://github.com/barbarabodinier/sharp/assets/92996311/0d1b9dd1-ed59-4b0f-afe1-4bc40283f65f">

<img width="711" alt="Screenshot 2023-10-27 at 16 02 16" src="https://github.com/barbarabodinier/sharp/assets/92996311/7ed6f184-fa21-4a61-9097-e2f3846ad9b2">

Hence the VariableSelection() is currently running a rank regression rather than a normal linear regression model for continuous outcomes. In case this wasn't intended the issue can easily be resolved by adding an additional if statement when turning vector into factor, and relocating the line where ydata is transformed into a matrix;

```
# Turning vector into factor
    if (**(family %in% c("binomial", "multinomial"))** & is.vector(ydata)) {
      ydata <- as.factor(ydata)
    }
    
    # Defining reference category and final data type
    if (is.factor(ydata)) {
    [...]
      ydata <- as.numeric(ydata) - 1
    }
    **ydata <- matrix(ydata, ncol = 1)**
  }
```

Interestingly, the resulting rank regression still performs really well, so I would potentially include it as an option for people to choose.

**Comment by barbarabodinier** [comment]
Thanks @ruweng for spotting this! Could you please open a pull request with the suggested changes?

**Comment by barbarabodinier** [comment]
The `ydata` object needs to be a matrix to allow for following steps to work.

**Comment by barbarabodinier** [comment]
This was fixed in commit bce0d55b3b3e4229f842cba10838edc2d9fad6df.
